File: app/src/main/java/com/android/score/MainActivity.java
package com.android.score;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import android.os.Bundle;
import androidx.lifecycle.ViewModelProvider;
import com.android.score.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {
    MyViewModel myViewModel;
    ActivityMainBinding binding;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this,R.layout.activity_main);
        myViewModel =  new ViewModelProvider.AndroidViewModelFactory(getApplication()).create(MyViewModel.class);
        binding.setData(myViewModel);
        binding.setLifecycleOwner(this);
    }
}

File: app/src/main/java/com/android/score/MainActivity.kt
package com.android.score

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.ViewModelProvider
import com.android.score.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        var binding:ActivityMainBinding = DataBindingUtil.setContentView(this,R.layout.activity_main)
        val  viewModel:MyViewModel = ViewModelProvider.AndroidViewModelFactory(application).create(MyViewModel::class.java)
        binding.data =viewModel
        binding.lifecycleOwner = this

    }
}